vm/systemSmartContracts/delegation.go

```diff
@@ -571,7 +571,14 @@ func (d *delegation) mergeValidatorDataToDelegation(args *vmcommon.ContractCallI
 		return vmcommon.UserError
 	}
 
-	return d.delegateUser(validatorData.TotalStakeValue, big.NewInt(0), validatorAddress, args.RecipientAddr, dStatus)
+	entry := &vmcommon.LogEntry{
```

Comment:
please replace all these (12) occurrences with a call to `createAndAddLogEntry`

Reply:
done.